scene/2d/canvas_item.cpp

```diff
@@ -1120,6 +1122,7 @@ void CanvasItem::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_edit_set_pivot", "pivot"), &CanvasItem::_edit_set_pivot);
 	ClassDB::bind_method(D_METHOD("_edit_get_pivot"), &CanvasItem::_edit_get_pivot);
 	ClassDB::bind_method(D_METHOD("_edit_use_pivot"), &CanvasItem::_edit_use_pivot);
+#endif
```

Comment:
This will break API portability. API must be the same for all platforms, targets and tools/non-tools. It's possible to register editor-only classes with `ClassDB::APIType`. We don't support this with individual methods/properties/signals/etc, only with classes..

Reply:
That's kind of annoying. If I understand correctly, it means there is no way to not compile those functions when tools are disabled ?
Using ClassDB::APIType is enough to make then not accessible, but well, that's wasted space in the executable.

Reply:
You may make it to not compile those functions with tools=no, but the methods binds must still be there.

Reply:
I don't think it is possible to bind a non-existing method. It will have compilation errors, unless we bind them as virtual but that is kind of hacky...

Reply:
You could `#ifdef` out the code *inside* the method definition, and add an `#else` statement that prints an error message when calling the method from a non-tools build

Do we have a standard way of doing this?